flaml/automl.py

```diff
@@ -943,10 +944,18 @@ def train(config: dict):
             if sample_size:
                 config['FLAML_sample_size'] = sample_size
             estimator = config['learner']
-            del config['learner']
-            result = states[estimator].training_function(config)
-            return result
-
+            # check memory constraints before training
+            if states[estimator].learner_class.size(config) <= MEM_THRES:

```

Comment:
mem_thres is a configurable option with default value MEM_THRES

Reply:
fixed. Thanks!